### PR TITLE
Preview: Adding Preliminary Video Inference Feature for Old And New Models

### PR DESCRIPTION
**Pull request by chancharikmitra**
This PR includes the following additions:

1. Support for video VQAScore inference for all previously-supported T2V models.
2. Addition of new interleaved-image/video LMM models.
3. Updating of the README and requirements in accordance with the additions

This PR enables preliminary support for video inference. More models, usability feature, etc. will be added in upcoming updates.

### CODE REVIEW

File: t2v_metrics/models/vqascore_models/gpt4v_model.py [hunk 1]
```diff
@@ -1,34 +1,49 @@
...
 import tiktoken
+import os
 
 from .vqa_model import VQAScoreModel
 
 default_question_template = 'Does this figure show "{}"? Please answer yes or no.'
 default_answer_template = 'Yes'
 
 GPT4V_MODELS = {
-    # We recommend using 'gpt-4-turbo' for optimal performance.
```
Comment (linzhiqiu) [inline t2v_metrics/models/vqascore_models/gpt4v_model.py:14]: Thanks for fixing this comment

File: t2v_metrics/models/vqascore_models/gpt4v_model.py [hunk 2]
```diff
@@ -44,97 +59,80 @@ def __init__(self,
...
-                completion = self.client.chat.completions.create(model=self.model_name, messages= [{"role": "user","content": [{"type": "text", "text": question},{ "type": "image_url","image_url": {"url": f"data:image/{image['type']};base64,{image['base64']}"}}]}],logprobs=True,top_logprobs=self.top_logprobs,)
-            except:
-                print(f"Failed image: {image['path']} and question: {question} and answer: {answer}")
-                return torch.Tensor([0.0])
-
-        # print(completion.choices[0].message)
-        # print(completion.choices[0].logprobs)
-        # print(completion.choices[0].logprobs.content[0])
-        is_generated = False
+        except Exception as e:
```
Comment (linzhiqiu) [inline t2v_metrics/models/vqascore_models/gpt4v_model.py:102]: The reason why I have it tried again was because sometimes ChatGPT just glitched -- but the second time you try it will work

File: t2v_metrics/models/vqascore_models/gpt4v_model.py [hunk 3]
```diff
@@ -44,97 +59,80 @@ def __init__(self,
...
+        
+        print(f"Warning: answer not generated for {data['path']} and question: {question}")
+        print(completion.choices[0].logprobs.content[0].top_logprobs)
+        return torch.Tensor([0.0])
 
-    # @torch.no_grad()
-    # @torch.autocast(device_type='cuda', dtype=torch.bfloat16)
     def forward(self,
-                images: List[str],
+                paths: List[str],
```
Comment (linzhiqiu) [inline t2v_metrics/models/vqascore_models/gpt4v_model.py:117]: Will this change be compatible with other higher-level modules?

File: t2v_metrics/models/vqascore_models/gpt4v_model_old.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
+from typing import List
```
Comment (linzhiqiu) [inline t2v_metrics/models/vqascore_models/gpt4v_model_old.py:1]: It is okay to delete this file

File: t2v_metrics/models/vqascore_models/instructblip_model.py [hunk 5]
```diff
@@ -132,6 +132,8 @@ def forward(self,
             labels=labels,
         )
         
+        print(labels.shape)
```
Comment (linzhiqiu) [inline t2v_metrics/models/vqascore_models/instructblip_model.py:135]: shall we remove the print statements?

File: t2v_metrics/models/vqascore_models/gpt4v_model.py (hunk 3 above)
Comment (linzhiqiu) [reply t2v_metrics/models/vqascore_models/gpt4v_model.py:117]: I guess we can stick with your new module -- if users want to pass in images then they use `images`, but if they want to pass in videos they can use `videos`. We can add a few assert statements to make sure the user didn't pass in both `images` and `videos`

File: README.md [hunk 6]
```diff
@@ -57,6 +57,36 @@ texts = ["someone talks on the phone angrily while another person sits happily",
...
+## (Preview) NEW! - Video Inference:
+
+This is a preview version of a new feature for T2V: video inference! Note that this feature requires installation of `flash-attn`. Please do this in addition to the evironment setup above to align with your system environment. 
+
+For single-image and CLIP-like models, video frames are concatenated. For all other native interleaved-image/video models (we recommend LLaVA-OneVision at the time of writing), video frames are passed directly to the model.
+
+```python
+import t2v_metrics
+
+### For a single (video, text) pair on a single-image model:
```
Comment (linzhiqiu) [inline README.md:69]: In addition to image-based VQAScore, I think it will be great to add 1-2 line demos for each of the new models you add (especially the real video VQAScore models)!

**Review by linzhiqiu — CHANGES_REQUESTED** [review]
Great job! I have a few comments to resolve before the merging.

File: t2v_metrics/models/vqascore_models/__init__.py [hunk 7]
```diff
@@ -25,8 +41,22 @@ def get_vqascore_model(model_name, device='cuda', cache_dir=HF_CACHE_DIR, **kwar
     elif model_name in LLAVA16_MODELS:
         return LLaVA16Model(model_name, device=device, cache_dir=cache_dir, **kwargs)
     elif model_name in InstructBLIP_MODELS:
-        return InstructBLIPModel(model_name, device=device, cache_dir=cache_dir, **kwargs)
+        return InstructBLIPModel(model_name, device=device, cache_dir=cache_dir)
```
[the remark this answers is not in the document]
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/__init__.py:44]: Fixed

File: .gitignore [hunk 8]
```diff
@@ -185,3 +185,11 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+# Wheels
+*.whl
+
+# Videos
```
[the remark this answers is not in the document]
Comment (chancharikmitra) [reply .gitignore:192]: Fixed

File: t2v_metrics/models/vqascore_models/__init__.py [hunk 9]
```diff
@@ -2,7 +2,15 @@
 from .llava_model import LLAVA_MODELS, LLaVAModel
 from .llava16_model import LLAVA16_MODELS, LLaVA16Model
 from .instructblip_model import InstructBLIP_MODELS, InstructBLIPModel
+from .qwenvl_model import QwenVL_MODELS, QwenVLModel
 from .gpt4v_model import GPT4V_MODELS, GPT4VModel
+from .llavaov_model import LLAVA_OV_MODELS, LLaVAOneVisionModel
+from .mplug_model import MPLUG_OWL3_MODELS, mPLUGOwl3Model
+from .paligemma_model import PALIGEMMA_MODELS, PaliGemmaModel
+from .internvl_model import INTERNVL2_MODELS, InternVL2Model
+from .internlm_model import INTERNLMXCOMPOSER25_MODELS, InternLMXComposer25Model
```
[the remark this answers is not in the document]
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/__init__.py:11]: Fixed

File: t2v_metrics/models/vqascore_models/instructblip_model.py (hunk 5 above)
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/instructblip_model.py:135]: Fixed

File: t2v_metrics/models/vqascore_models/clip_t5_model.py [hunk 10]
```diff
@@ -277,6 +277,8 @@ def forward(self,
         lm_prob = torch.zeros(logits.shape[0])
         loss_fct = torch.nn.CrossEntropyLoss(reduction='mean')
         for k in range(lm_prob.shape[0]):
+            print(logits.shape)
```
[the remark this answers is not in the document]
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/clip_t5_model.py:280]: Fixed

File: t2v_metrics/models/vqascore_models/gpt4v_model.py (hunk 3 above)
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/gpt4v_model.py:117]: Yes, this should work fine since whether it is `images` or `videos` passed by the user, the preprocessing is done at the model level.

File: t2v_metrics/models/vqascore_models/gpt4v_model_old.py (hunk 4 above)
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/gpt4v_model_old.py:1]: Fixed

File: t2v_metrics/models/vqascore_models/gpt4v_model.py (hunk 2 above)
Comment (chancharikmitra) [reply t2v_metrics/models/vqascore_models/gpt4v_model.py:102]: Fixed

File: mplug_requirements.txt [hunk 11]
```diff
@@ -0,0 +1,23 @@
+--extra-index-url https://mirrors.cloud.tencent.com/pypi/simple/
+icecream
+markdown2
+modelscope
+pydantic
+accelerate
+transformers==4.37.2
```
[the remark this answers is not in the document]
Comment (chancharikmitra) [reply mplug_requirements.txt:7]: Yes, I carefully checked and stitched together requirements from T2V, LLaVA-NeXT, and MPLUG that all work together.

File: README.md (hunk 6 above)
Comment (chancharikmitra) [reply README.md:69]: Sorry, I think the typos made it seem like I was just using CLIP-FlanT5, but I also do a representative example for LLaVA-OV. The others that are multi-image like mPLUG or InternVL2 have the same usage.

File: test.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
+import t2v_metrics
```
Comment (linzhiqiu) [inline test.py:1]: Shall we delete this file?

**Review by linzhiqiu — CHANGES_REQUESTED** [review]
Minor issues
